tests/other/test_smtlibv2.py

```diff
@@ -26,82 +27,362 @@
+
+        check(BitvecShiftLeft, operanda=bvx, operandb=bvy, pickle_size=136, sizeof=56)
+        check(BitvecShiftRight, operanda=bvx, operandb=bvy, pickle_size=137, sizeof=56)
+        check(BitvecArithmeticShiftLeft, operanda=bvx, operandb=bvy, pickle_size=146, sizeof=56)
+        check(BitvecArithmeticShiftRight, operanda=bvx, operandb=bvy, pickle_size=147, sizeof=56)
+
+        check(BitvecZeroExtend, operand=bvx, size=122, pickle_size=119, sizeof=56)
+        check(BitvecSignExtend, operand=bvx, size=122, pickle_size=119, sizeof=56)
+        check(BitvecExtract, operand=bvx, offset=0, size=8, pickle_size=119, sizeof=64)
+        check(BitvecConcat, operands=(bvx, bvy), pickle_size=133, sizeof=56)
+        check(BitvecITE, condition=x, true_value=bvx, false_value=bvy, pickle_size=161, sizeof=56)
+
+        a = ArrayVariable(index_size=32, value_size=32, length=324, name="name")
+        check(ArrayConstant, index_size=32, value_size=8, value=b"A", pickle_size=136, sizeof=64)
+        check(ArraySlice, array=a, offset=0, size=10, pickle_size=136, sizeof=48)
+        check(ArraySelect, array=a, index=bvx, pickle_size=161, sizeof=56)
+        check(ArrayStore, array=a, index=bvx, value=bvy, pickle_size=188, sizeof=80)
+
+        def all_subclasses(cls):
+            return set((Expression,)).union(
+                set(cls.__subclasses__()).union(
+                    [s for c in cls.__subclasses__() for s in all_subclasses(c)]
+                )
+            )
 
-"""
-class Z3Specific(unittest.TestCase):
+        all_types = all_subclasses(Expression)
+        self.assertSetEqual(checked, all_types)
+
+    def test_Expression_BitvecOp(self):
+        a = BitvecConstant(size=32, value=100)
+        b = BitvecConstant(size=32, value=101)
+        x = a + b
+        self.assertTrue(isinstance(x, Bitvec))
+
+    def test_Expression_BoolTaint(self):
+        # Bool can not be instantiaated
+        self.assertRaises(Exception, Bool, ())
+
+        x = BoolConstant(value=True, taint=("red",))
+        y = BoolConstant(value=False, taint=("blue",))
+        z = BoolOr(x, y)
+        self.assertIn("red", x.taint)
+        self.assertIn("blue", y.taint)
+        self.assertIn("red", z.taint)
+        self.assertIn("blue", z.taint)
+
+    def test_Expression_BitvecTaint(self):
+        # Bool can not be instantiaated
+        self.assertRaises(Exception, Bitvec, ())
+
+        x = BitvecConstant(size=32, value=123, taint=("red",))
+        y = BitvecConstant(size=32, value=124, taint=("blue",))
+        z = BoolGreaterOrEqualThan(x, y)
+        self.assertIn("red", x.taint)
+        self.assertIn("blue", y.taint)
+        self.assertIn("red", z.taint)
+        self.assertIn("blue", z.taint)
+
+    def test_Expression_Array(self):
+        # Bool can not be instantiaated
+        self.assertRaises(Exception, Array, ())
+
+        a = ArrayConstant(index_size=32, value_size=8, value=b"ABCDE")
+        a[0] == ord("A")
```

Comment:
This statement has no effect? Are we trying to assert that `a[0]` is equal to `ord("A")`?